test/FunctionalTests/DeadlineTests.cs

```diff
@@ -90,5 +90,84 @@ public async Task WriteUntilDeadline_SuccessResponsesStreamed(string requestUri)
             Assert.AreEqual(StatusCode.DeadlineExceeded.ToTrailerString(), Fixture.TrailersContainer.Trailers[GrpcProtocolConstants.StatusTrailer].Single());
             Assert.AreEqual("Deadline Exceeded", Fixture.TrailersContainer.Trailers[GrpcProtocolConstants.MessageTrailer].Single());
         }
+
+        [Test]
+        public async Task WriteMessageAfterDeadline()
+        {
+            static async Task WriteUntilError(HelloRequest request, IServerStreamWriter<HelloReply> responseStream, ServerCallContext context)
+            {
+                var i = 0;
+                while (true)
+                {
+                    var message = $"How are you {request.Name}? {i}";
+                    await responseStream.WriteAsync(new HelloReply { Message = message });
+                    i++;
+
+                    await Task.Delay(10);
```

Comment:
I'm wary of task delays in tests. Looks like in this case, it's just to reduce the iteration speed since the timeout is set to 200ms?

Reply:
Yeah. No need to blast through messages as quickly as possible.